presto-main/src/main/java/com/facebook/presto/operator/scalar/annotations/ScalarImplementation.java

```diff
@@ -337,6 +338,12 @@ private void parseArguments(Method method)
                         }
                     }
 
+                    // skip argument type check for CodePoints
+                    if (!type.value().equals(CodePointsType.NAME)) {
```

Comment:
This is hacky. This method/class shouldn't need to be aware of any concrete types. Why is this needed?